stubs/python-gnupg/gnupg/__init__.pyi

```diff
@@ -0,0 +1,257 @@
+import logging
+import re
+from typing import Any
+
+__version__: str = ...
+__author__: str = ...
+__date__: str = ...
+
+from typing import Protocol
+
+string_types: type[str] = ...
+text_type: type[str] = ...
+path_types: tuple[type[str]] = ...
+
+log_everything: bool = ...
+logger: logging.Logger = ...
+fsencoding: str = ...
+
+UNSAFE: re.Pattern[Any] = ...
+
+def shell_quote(s: str) -> str: ...
+def no_quote(s: str) -> str: ...
+
+class _ReadableFile(Protocol):
+    def read(self, size: int) -> bytes: ...
```

Comment:
A few issues:

* In some functions, `read()` is called with a `size` argument.
* The `size` argument is only called positionally, so its name doesn't matter.
* `close()` is also called on the file.

```suggestion
class _ReadableFile(Protocol):
    def read(self, size: int = ..., /) -> bytes: ...
    def close(self) -> object: ...
```